cagov/ui-claim-tracker
# Add ability to have language-dependent urls

## Ticket

Resolves #324 

## Changes

- Move language-dependent urls into `common.json`
- Move url resolution logic from `getClaimStatus` to `<TransLine>`
- Update scenario content to latest content from UIB
- Move logic to determine whether the user came from UIO Mobile from the `<Home>` component to `getServerSideProps()`
- Add `userArrivedFromUioMobile` argument to many components
- Split `TransLineContent` type from `TransLineProps` type to better mirror other types in `common`
- Add `userArrivedFromUioMobile` boolean control to the Page story

## Context

This PR has a lot of changes, but they can be loosely organized into three buckets:

1. Changes to support language-dependent url resolution
2. Changes to support mobile vs non-mobile url resolution
3. Renaming types

## Testing

`yarn test`

PLUS

While #299 is not resolved, comment out the following lines, then run `yarn storybook`, and verify that the links to UIO mobile work when the toggle is set to `true`: https://github.com/cagov/ui-claim-tracker/blob/32990f892c6b84017c30af5d273fdd94410649e7/pages/index.tsx#L95-L98

PLUS

1. Run `yarn dev`
2. Load localhost:3000
3. Click "En español"
4. Make sure the links change to their `-espanol` version
